client-index/src/synchronizer.rs

```diff
@@ -48,37 +48,44 @@ where
         private_key: &PrivateKey,
         batch_size: Option<usize>,
     ) -> Result<()> {
+        let status = self.client.status()?;
+
         let last_block_height = self.global_state_service.last_block_height(view_key)?;
-        let current_block_height = self.client.status()?.last_block_height()?;
+        let current_block_height = status.last_block_height()?;
 
         // Send batch RPC requests to tendermint in chunks of `batch_size` requests per batch call
         for chunk in ((last_block_height + 1)..=current_block_height)
             .chunks(batch_size.unwrap_or(DEFAULT_BATCH_SIZE))
             .into_iter()
         {
+            if self.fast_forward_status(staking_addresses, view_key, private_key, &status)? {
```

Comment:
would it be possible to give a command-line option, e.g. `--fast` or `--slow`, so that one can still do the previous block-by-block check without the fast-forward?

Reply:
Is there any reason why someone would want to do that?

Reply:
for sanity